packages/admin-frontend/pages/crop.js

```diff
@@ -0,0 +1,188 @@
+  imageApiBody?: any,
+  existingParameters?: any,
+  postResult?: string
+};
+
+class Crop extends Component<Props, State> {
+  state = { ratio: 0.81, imageApiBody: null, existingParameters: null };
+
+  toPercentages = c => {
+    var cropStartPxX = c.getCropBoxData().left - c.getCanvasData().left;
+    var cropEndPxX = cropStartPxX + c.getCropBoxData().width;
+    var cropStartPxY = c.getCropBoxData().top - c.getCanvasData().top;
+    var cropEndPxY = cropStartPxY + c.getCropBoxData().height;
+
+    return {
+      cropStartX: Math.max(
+        0,
+        Math.round((cropStartPxX / c.getImageData().naturalWidth) * 100)
+      ),
+      cropEndX: Math.min(
+        100,
+        Math.round((cropEndPxX / c.getImageData().naturalWidth) * 100)
+      ),
+      cropStartY: Math.max(
+        0,
+        Math.round((cropStartPxY / c.getImageData().naturalHeight) * 100)
+      ),
+
+      cropEndY: Math.min(
+        100,
+        Math.round((cropEndPxY / c.getImageData().naturalHeight) * 100)
+      )
+    };
+  };
+
+  toImageApiBody = pcnt => {
+    const existingParametersForCurrentRatio =
+      this.state.existingParameters &&
+      this.state.existingParameters.find(
+        p => p.forRatio === String(this.state.ratio)
+      );
+    const revision =
+      (existingParametersForCurrentRatio &&
+        existingParametersForCurrentRatio.revision) ||
+      1;
+
+    return {
+      forRatio: String(this.state.ratio),
+      revision: revision,
+      imageUrl:
+        this.props.imageUrl &&
+        this.props.imageUrl.substr(this.props.imageUrl.lastIndexOf('/')),
+      rawImageQueryParameters: {
+        cropStartX: pcnt.cropStartX,
+        cropEndX: pcnt.cropEndX,
+        cropStartY: pcnt.cropStartY,
+        cropEndY: pcnt.cropEndY
+      }
+    };
+  };
+
+  _crop() {
+    var pcnt = this.toPercentages(this.refs.cropper);
+    this.setState({ imageApiBody: this.toImageApiBody(pcnt) });
+  }
+
+  postToImageApi = async _ => {
+    if (this.state.imageApiBody !== null) {
+      this.setState({ postResult: 'Posting…' });
+      const response = await fetch(`${imageApiUrl}/images/stored-parameters`, {
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer ' + getTokenFromLocalCookie(),
+          Accept: 'application/json'
+        },
+        body: JSON.stringify(this.state.imageApiBody)
+      });
+
+      if (response.ok) {
+        this.setState({ postResult: 'Ok!' });
+      } else if (response.status === 401) {
+        this.setState({ postResult: 'Unauthorized, please login' });
+      } else {
+        this.setState({ postResult: `Unknown error: ${response.status}` });
+      }
+      await this.getExistingParameters();
+    }
+  };
+
+  getExistingParameters = async () => {
+    const imageUrl =
+      this.props.imageUrl &&
+      this.props.imageUrl.substr(this.props.imageUrl.lastIndexOf('/'));
+    const url =
+      imageUrl &&
+      `${imageApiUrl}/images/stored-parameters${imageUrl.toString()}`;
```

Comment:
is toString() necessary here?

Reply:
No, I misunderstood a Flow error I got earlier.